TypeSystem/variant.h

```diff
@@ -40,6 +40,32 @@ SOFTWARE.
 
 namespace current {
 
+// A high-octane version of `current::metaprogramming::combine.
+namespace variant_high_octane_tmp {
+
+template <typename T, typename... TS>
+struct fast_combine : fast_combine<T>, fast_combine<TS...> {
+  using fast_combine<T>::operator();
+  using fast_combine<TS...>::operator();
+};
+
+template <typename T>
+struct fast_combine<T> {
+  T instance;
+
+  template <typename PARAM>
+  void operator()(typename T::first_parameter_1 first, PARAM&& param, typename T::third_parameter& third) {
+    instance.operator()(first, std::forward<PARAM>(param), third);
+  }
+
+  template <typename PARAM>
+  void operator()(typename T::first_parameter_2 first, PARAM&& param, typename T::third_parameter& third) {
+    instance.operator()(first, std::forward<PARAM>(param), third);
+  }
+};
```

Comment:
Without `using`-s it won't compile on g++, only on clang++. Just checked.

And on clang++, the difference in compilation time is negligible.